src/meta/meta_turtle.c

```diff
@@ -80,129 +92,180 @@ __metadata_backup_target_uri_search(
         return (0);
-    WT_RET(__wt_fopen(session, WT_METADATA_BACKUP, 0, WT_STREAM_READ, &fs));
+    WT_RET(__wt_fopen(session, file, 0, WT_STREAM_READ, &fs));
 
-    /* Read line pairs and load them into the metadata file. */
-    WT_ERR(__wt_scr_alloc(session, 512, &key));
-    WT_ERR(__wt_scr_alloc(session, 512, &value));
+    /* Read line pairs and add them to the import list. */
+    WT_ERR(__wt_scr_alloc(session, 1024, &key));
+    WT_ERR(__wt_scr_alloc(session, 1024, &value));
     for (;;) {
         WT_ERR(__wt_getline(session, fs, key));
         if (key->size == 0)
             break;
         WT_ERR(__wt_getline(session, fs, value));
         if (value->size == 0)
-            WT_ERR_PANIC(session, EINVAL, "%s: zero-length value", WT_METADATA_BACKUP);
-        /*
-         * When performing partial backup restore, generate a list of tables that is not part of the
-         * target uri list so that we can drop all entries later. To do this, parse through all the
-         * table metadata entries and check if the metadata entry exists in the target uri hash
-         * table. If the metadata entry doesn't exist in the hash table, append the table name to
-         * the partial backup remove list.
-         */
-        metadata_key = (char *)key->data;
-        if (F_ISSET(conn, WT_CONN_BACKUP_PARTIAL_RESTORE) &&
-          WT_PREFIX_MATCH(metadata_key, "table:")) {
-            /* Assert that there should be no WiredTiger tables with a table format. */
-            WT_ASSERT(
-              session, __wt_name_check(session, (const char *)key->data, key->size, true) == 0);
-            /*
-             * The target uri will be the deciding factor if a specific metadata table entry needs
-             * to be dropped. If the metadata table entry does not exist in the target uri hash
-             * table, append the metadata key to the backup remove list.
-             */
-            if (__metadata_backup_target_uri_search(session, backuphash, metadata_key) == false) {
-                if (key->size > max_len)
-                    max_len = key->size;
-                WT_ERR(__wt_realloc_def(session, &allocated_name, slot + 2, &partial_backup_names));
-                p = &partial_backup_names[slot];
-                p[0] = p[1] = NULL;
-
-                WT_ERR(
-                  __wt_strndup(session, (char *)key->data, key->size, &partial_backup_names[slot]));
-                slot++;
-            }
-        }
+            WT_ERR_PANIC(session, EINVAL, "%s: zero-length value", file);
+
+        WT_ERR(md_entry_worker_func(session, key, value, state));
+    }
+
+err:
+    __wt_scr_free(session, &key);
+    __wt_scr_free(session, &value);
+    WT_TRET(__wt_fclose(session, &fs));
 
+    return (ret);
+}
+
+/*
+ * __metadata_entry_worker --
+ *     Worker function for metadata file reader procedure. The function populates partial backup
+ *     names array and updates the metadata of database with the new entries read from the file.
```

Comment:
Minor wording suggestion (add 'the' in two places):
... The function populates the backup names array... the metadata of the database...

Reply:
Done.